Plugins/JiraCommitTipPlugin/JiraComminTipPlugin.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Windows.Forms;
+using Atlassian.Jira;
+using GitUI.CommandsDialogs;
+using GitUIPluginInterfaces;
+using JiraCommitPlugin;
+using ResourceManager;
+
+namespace JiraComminTipPlugin
+{
+    public class JiraComminPlugin : GitPluginBase, IGitPluginForRepository
+    {
+        private const string description = "Jira Commit Tip";
+        private Jira jira;
+        private JiraNamedEntity filter;
+        private readonly StringSetting url = new StringSetting("Jira URL", @"https://jira.atlassian.com");
+        private readonly StringSetting user = new StringSetting("Jira user", string.Empty);
+        private readonly PasswordSetting password = new PasswordSetting("Jira password");
```

Comment:
What about paranoic-mode when password for jira account isn't saved to the disc and must be entered on each app run? I'm not very comfortable having password saved in plain text.